diff --git a/src/gen_leader.erl b/src/gen_leader.erl
index fe70d99..d554e50 100644
--- a/src/gen_leader.erl
+++ b/src/gen_leader.erl
@@ -531,6 +531,9 @@ init_it(Starter,Parent,Name,Mod,{UnsortedCandidateNodes,OptArgs,Arg},Options) ->
 % is complete.
 safe_loop(#server{mod = Mod, state = State} = Server, Role,
           #election{name = Name} = E, _PrevMsg) ->
+    morpheus_guest:raw_apply(ets, delete, [test_state_tab, {state, self()}]),
+    morpheus_guest:raw_apply(ets, insert, [test_state_tab, {{state, self()}, {safe_loop, Server, Role, Name}}]),
+    morpheus_guest:report_state(lists:sort(morpheus_guest:raw_apply(ets, match, [test_state_tab, '$1']))),
     receive
         {system, From, Req} ->
             #server{parent = Parent, debug = Debug} = Server,
@@ -805,6 +808,9 @@ loop(#server{parent = Parent,
              state = State,
              debug = Debug} = Server, Role,
      #election{name = Name} = E, _PrevMsg) ->
+    morpheus_guest:raw_apply(ets, delete, [test_state_tab, {state, self()}]),
+    morpheus_guest:raw_apply(ets, insert, [test_state_tab, {{state, self()}, {loop, Server, Role, Name}}]),
+    morpheus_guest:report_state(lists:sort(morpheus_guest:raw_apply(ets, match, [test_state_tab, '$1']))),
     receive
         Msg ->
             case Msg of
